presentation/src/main/java/com/pomonyang/mohanyang/presentation/screen/statistics/component/graph/FocusTimeGrpahBox.kt

```diff
@@ -118,14 +147,36 @@ private fun FocusGraphXAxis(
                     MnTheme.iconColorScheme.secondary
                 }
 
+                val height = animateDpAsState(
+                    targetValue = if (barAnimated) barHeight else 0.dp,
+                    animationSpec = tween(
+                        durationMillis = 1000,
+                        delayMillis = idx * 100, // 각 바에 대한 지연 시간
+                        easing = FastOutSlowInEasing,
+                    ),
+                    finishedListener = { barAnimated = true },
+                    label = "barHeight",
+                )
+
+                val color by animateColorAsState(
+                    targetValue = if (barAnimated) barColor else Color.Transparent,
+                    tween(
+                        durationMillis = 2000,
+                        delayMillis = idx * 100, // 각 바에 대한 지연 시간
+                        easing = FastOutSlowInEasing,
+                    ),
+                )
+
+                val isMaxFocusTime = data > 0 && data == maxFocusTime
```

Comment:
```suggestion
val isMaxFocusTime = remember(data, maxFocusTime) { data > 0 && data == maxFocusTime }
```

애니메이션으로 인해 recomposition 많이 탈거 같아서 이렇게 관리하면 더 좋을 것 같네

Reply:
graphData가 변경 될때만 recomposition이 될 것 같은데 어떨 때 애니메이션 변화로 recomposition 될까?..?
나도 첨에 remember 해야하나 생각했는데 이 bar를 그릴 때 판단해야하는 flag라 생각해서 논리비교로 남기긴 했는데 🧐

Reply:
```kotlin
val color by animateColorAsState(
    targetValue = if (barAnimated) barColor else Color.Transparent,
    tween(
        durationMillis = 2000,
        delayMillis = idx * 100, // 각 바에 대한 지연 시간
        easing = FastOutSlowInEasing,
    ),
)

println("recomposition $idx")
```

color 밑에 이렇게 println을 찍어보면 애니메이션 변경 값만큼 찍히는데, `animateAsState` 로 만든 순간 저 Composable Group이 Recomposition 대상이라서 계속 타게 되는 상황이라서 논리 비교를 계속하게 되는데 불필요해 보였어~!

Reply:
아 animateStatate 는 생각 못했다! 알려줘서 감사합니다  bb